app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -298,20 +305,47 @@ private void updateNavigationMenuSelection() {
         }
     }
 
-    private void addRoomColumns(HorizontalSnapScrollView scroller) {
-        LinearLayout root = (LinearLayout) scroller.getChildAt(0);
-        root.removeAllViews();
-        if (scroller.getColumnWidth() != 0) {
-            // update pre-calculated width with actual layout
-            columnWidth = scroller.getColumnWidth();
+    private void addRoomColumns(@NonNull HorizontalSnapScrollView horizontalScroller, @NonNull List<Lecture> lectures, int roomCount, boolean forceReload) {
+        int roomIndexLeft = horizontalScroller.getColumn();
+        int roomIndexRight = roomIndexLeft + maxRoomColumnsVisible;
+
+        // whenever possible, just update recycler views
+        if (!forceReload && !adapterByRoom.isEmpty()) {
+            for (int roomIndex = roomIndexLeft; roomIndex < roomIndexRight; roomIndex++) {
+                //noinspection ConstantConditions
+                adapterByRoom.get(roomIndex).notifyDataSetChanged();
```

Comment:
:bomb: This line crashes the application when I return from the details screen to the main screen after favoring a lecture:

``` java
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 
'void nerd.tuxmobil.fahrplan.congress.schedule.LectureViewColumnAdapter.notifyDataSetChanged()' 
on a null object reference
    at nerd.tuxmobil.fahrplan.congress.schedule.FahrplanFragment.addRoomColumns(FahrplanFragment.java:316)
    at nerd.tuxmobil.fahrplan.congress.schedule.FahrplanFragment.viewDay(FahrplanFragment.java:283)
    at nerd.tuxmobil.fahrplan.congress.schedule.FahrplanFragment.onResume(FahrplanFragment.java:243)
    at android.support.v4.app.Fragment.performResume(Fragment.java:2498)
```

Reply:
I've fixed that one. Thanks for spotting. It was due to sparse room indices.

Reply:
The app still crashes with a7d74866c26323dbe98b6df03748517b68088adf if I swipe to the last "Assembly:IT-Kollektiv" room on day 1. Also, the room title appears twice and contains the lectures of "Hackers Beauty Palace". It looks like there is a off by one error which mixes up room titles and lectures.

Reply:
Not crashing anymore with d25b62930434d0badafc0eea7ebd1ff919b1f21c. Closing here.